x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -306,6 +373,31 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
     return this._store;
   }
 
+  /**
+   * Lazily instantiate a `SecurityAppStore` for discover.
+   */
+  private async getStoreForDiscover(
```

Comment:
Why do we need to create a separate store for Discover? Is there any reason not to share `this._store`? 
If we use two different stores we may end up with inconsistent states between apps.

Reply:
> If we use two different stores we may end up with inconsistent states between apps.

In my opinion, that is exactly what we need, we don't want the state of Security affecting what is happening in discover and vice versa. Consider below scenarios : 

1. If a row actions changes some state and user performs that in discover, we do not want that to be affected in Security solution. For example, we set lot of global queries in Security solution and row actions do 2 things related to that:
    - execute `refetch` provided by consumer such as `timeline`, `alert-table` and in this case `discover`.
    - execute `reftechGlobalQuery`
    We do not want discover to be affacted by global query re-fetches since it is a separate app and should not be triggering those re-fetches.


2. Discover is completely separate app which results in different values in KQL query bar, global filters and timeranges. Security App state observes those ( using data plugin ) and updates the state based on those. We do not want that as well. Consider user adds a filter in discover, then that filter changes the state of Security App.

![grafik](https://github.com/user-attachments/assets/9152b907-69db-4160-81fb-10887d375e73)


I understand that there is a workaround of these kind of scenarios but i think it is much safer to keep State completely separate for Discover since it is a different app and not just a different page.

Reply:
> I assumed this task aimed to make it a more integrated part of Security Solution, for the user, despite being a separate plugin.

I do not think so. I think it is a product question. For now, we just want to give same UX experience but Discover continues to be different app working on its own.

>Right now if we add a filter or search query anywhere in Security it persists across the application

Not in Discover. I checked. See below video. And it makes sense because Discover query bar provides different set of features ( mainly ESQL ) than global query bar.


https://github.com/user-attachments/assets/3f79ad13-e17a-4530-9759-69654a5b574d

>I guess that's

I guess you left your thought midway 🤣 .

Reply:
😂  yes, edited.

Okay, I see, but I would like to know the long-term vision because it would also make sense to be shared.
Let's discuss that with @paulewing

Reply:
>I was also thinking about security-specific actions, let's imagine in the flyout we have an action to "add to timeline" (I know the timeline will probably go away, but it's an easy example), with the current approach, we would be adding it to the discover store, when we go back to security anything would have changed. 

I am not sure for now if we have any use case which affects Security Solution state from Discover. Ideally, actions like that should not exist.

Reply:
@michaelolo24 , do you have any thoughts here as well?

Reply:
Hey @logeekal - made the comment separately, but the simplest solution moving forward would be best, which in this case would be having a separate state wrapper for discover. I fully expect that based on what features we actually implement there, we will end up changing this, or removing it entirely in favor of state sharing through the url or localstorage. For now, this allows us to make progress, and provides a system for sharing state if we find the url/browser storage options don't end up satisfying our long term goals. 

If this does end up being the best route forward, we can think through having a shared state then and how to properly isolate the slices of state that discover needs to know about. Thanks all!

Reply:
Thanks @michaelolo24.

@semd , Do you mind finalizing your review on this PR and let me know if there are any more blocker.